vnext/ReactUWP/Modules/AsyncStorageModuleWin32.cpp

```diff
@@ -0,0 +1,191 @@
+#include "pch.h"
+
+#include "AsyncStorageModuleWin32.h"
+
+#include <cstdio>
+
+/// Implements ASM using winsqlite3.dll (requires Windows version 10.0.10586)
+
+namespace facebook {
+namespace react {
+
+namespace {
+
+static void ExecImpl(sqlite3 *db, const char *statement, AsyncStorageModuleWin32::ExecCallback callback, void *pv) {
+  char *errMsg = nullptr;
+  int rc = sqlite3_exec(db, statement, callback, pv, &errMsg);
+  if (errMsg) {
+    throw std::runtime_error(errMsg);
+  }
+  if (rc != SQLITE_OK) {
+    throw std::runtime_error(sqlite3_errmsg(db));
+  }
+}
+
+class Sqlite3Transaction {
+  sqlite3 *m_db{nullptr};
+
+ public:
+  explicit Sqlite3Transaction(sqlite3 *db) : m_db(db) {
+    if (db) {
+      ExecImpl(db, u8"BEGIN TRANSACTION", nullptr, nullptr);
+    }
+  }
+  Sqlite3Transaction(const Sqlite3Transaction &) = delete;
+  Sqlite3Transaction(Sqlite3Transaction &&other) : m_db{other.m_db} {
+    other.m_db = nullptr;
+  }
+  Sqlite3Transaction &operator=(const Sqlite3Transaction &) = delete;
+  Sqlite3Transaction &operator=(Sqlite3Transaction &&rhs) {
+    std::swap(m_db, rhs.m_db);
```

Comment:
>std::swap [](start = 4, length = 9)

It does not seem right: rhs.m_db must be set to nullptr and method must return a value: *this;

Reply:
Yeah it turns out that donating blood is not conducive to writing good C++. 🧛‍♀️